plugins/default/src/Ide/LocalBindings.hs

```diff
@@ -0,0 +1,105 @@
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase                 #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+
+module Ide.LocalBindings
+  ( Bindings
+  , getLocalScope
+  , mostSpecificSpan
+  , holify
+  , bindings
+  ) where
+
+import           Data.Function
+import           Data.Generics
+import           Data.IntervalMap.FingerTree (IntervalMap, Interval (..))
+import qualified Data.IntervalMap.FingerTree as IM
+import           Data.List
+import qualified Data.Map as M
+import           Data.Maybe
+import           Data.Ord
+import           Data.Set (Set)
+import qualified Data.Set as S
+import           Development.IDE.GHC.Compat (GhcTc, RefMap, identType, noExt)
+import           HsExpr
+import           Id
+import           OccName
+import           SrcLoc
+
+------------------------------------------------------------------------------
+-- | Turn a 'RealSrcSpan' into an 'Interval'.
+realSrcSpanToInterval :: RealSrcSpan -> Interval RealSrcLoc
+realSrcSpanToInterval rss =
+  Interval
+    (realSrcSpanStart rss)
+    (realSrcSpanEnd rss)
+
+
+------------------------------------------------------------------------------
+-- | Compute which identifiers are in scpoe at every point in the AST. Use
+-- 'getLocalScope' to find the results.
+bindings :: RefMap -> Bindings
+bindings refmap = Bindings $ foldr (uncurry IM.insert) mempty  $ do
+  (ident, refs) <- M.toList refmap
+  Right name <- pure ident
+  (ref_span, ident_details) <- refs
+  Just ty <- pure $ identType ident_details
+  pure ( realSrcSpanToInterval ref_span
```

Comment:
I don't think this is what you want. The `ref_span` is the span of each occurrence of `name`. I think what is needed is the `Scope` obtained by calling [`getScopeFromContext`](https://hackage.haskell.org/package/ghc-8.10.1/docs/HieUtils.html#v:getScopeFromContext) on `identInfo`.

Reply:
Thanks, fixed!